sim/simulator.py

```diff
@@ -398,7 +400,7 @@ def on_message(self, msg: Message) -> None:
                     "sim://uni@server",
                     CreatePromiseWithTaskReq(
                         id=id,
-                        timeout=conv.timeout,
+                        timeout=int((self.time + conv.timeout) * 1000),
```

Comment:
Couldn't we use `time_ms(self.time, con.timeout)` here?

Reply:
I have removed the helper function, I think it made things more confusing